### ISSUE TITLE
Table header cells created as `<td>` elements

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#2431]
### Description

When a user creates a new table in Pressbooks and designates a row as the table header, the cells within that row remain `<td>` elements. They should be `<th>`. See https://www.w3.org/MarkUp/html3/tablecells.html#:~:text=TH%20is%20used%20for%20table,heavier%20font%20for%20header%20cells.&text=The%20CLASS%20attribute%20can%20be,instance%20into%20heads%20and%20subheads.

### Steps to Reproduce

1. Create a new table in your book
2. Select the top row and designate it as the table header
3. View the HTML output for your book and observe that cells inside the `<thead>` element are rendered as `<td>` elements rather than the preferred `<th>` elements.

https://user-images.githubusercontent.com/13485451/137010335-0e50c881-6072-43ad-a1dc-d10467551c67.mp4

**Expected behavior:** 
When a user designates a row as a table header, the cells in that row should be rendered as `<th>` elements.

**Actual behavior:** 
When a user designates a row as a table header, the cells in that row are rendered as `<td>` elements.

## Notes
We use the build in TinyMCE table editor (with some customizations). See https://github.com/pressbooks/pressbooks/blob/16e57f7f97e88ba5c74b2121d7e07960b14cbe10/inc/editor/namespace.php#L305-L395

**Comment by SteelWagstaff** [comment]
Check https://www.tiny.cloud/docs/plugins/opensource/table/ and do further research.

**Comment by SteelWagstaff** [comment]
It appears that accessible headings can be created using TinyMCE 5.4 and later: https://www.tiny.cloud/docs/plugins/opensource/table/#table_header_type

**Comment by SteelWagstaff** [comment]
WordPress uses TinyMCE [4.9.11] (https://github.com/WordPress/WordPress/commit/e503ee45d60171b0ecaf16a615a079a25c54ac55) and has no plans to update anytime soon: https://core.trac.wordpress.org/ticket/47218#comment:7. This version of TinyMCE has been out of support since the end of 2020. The current version of TinyMCE is 5.10. Recent versions of TinyMCE include several improvements that would be useful for WordPress users who continue to use the Classic Editor rather than Gutenberg, including several accessibility improvements and security fixes (see https://www.tiny.cloud/docs/release-notes/ for specific details). I don't know how feasible it is for us to update the version of TinyMCE that we use WordPress' Classic editor makes use of TinyMCE. This guide includes relevant details for projects which intend to upgrade from v4 to v5: https://www.tiny.cloud/docs/migration-from-4x/.

**Comment by MichSpieler** [comment]
We have had several support requests since the launch of Editoria11y where users report this either as a bug or ask what they need to do. This would be great to solve with the accessibility deadlines coming up.

**Comment by SteelWagstaff** [comment]
I agree. I think this could be a fairly simple fix. Will try to open a PR to address this weekend.

**Comment by SteelWagstaff** [comment]
I opened a PR to address this: https://github.com/pressbooks/pressbooks/pull/4304

**Comment by tw77** [comment]
<img width="720" height="421" alt="Image" src="https://github.com/user-attachments/assets/123defa2-325f-4ad6-8641-d86d96d2863d" />

Working as expected for me on dev, ready for release.
